iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -1840,7 +1841,8 @@ private List<String> getPartitions(DDLOperationContext context, Configuration jo
       fetcher.initialize(job, HiveTableUtil.getSerializationProps());
       org.apache.hadoop.hive.ql.metadata.Table metaDataPartTable =
           context.getDb().getTable(hmstbl.getDbName(), hmstbl.getTableName(), "partitions", true);
-      Deserializer currSerDe = metaDataPartTable.getDeserializer();
+      Deserializer currSerDe = HiveMetaStoreUtils.getDeserializer(job, metaDataPartTable.getTTable(),
```

Comment:
Use job conf to get the iceberg Deserializer so that we can propagate `ColumnProjectionUtils.READ_COLUMN_NAMES_CONF_STR` to `HiveIcebergSerDe`
https://github.com/apache/hive/blob/e570e65a397aae47c62472f12acaaed06b68cf54/iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergSerDe.java#L163